src/shared/main/java/com/couchbase/lite/internal/core/C4BlobWriteStream.java

```diff
@@ -55,7 +55,14 @@
      * Writes data to a stream.
      */
     public void write(byte[] bytes) throws LiteCoreException {
-        write(handle, bytes);
+        write(handle, bytes, 0);
```

Comment:
Would this be better as `write(handle, bytes, bytes.getLength()` ?  That would simplify the whole "0 means everything" logic down in JNI.

Reply:
I will do that.  It is a good idea.  I can't change the constructor, though (it needs the default 0 and its odd meaning) because there are lots of other places that use it.

Reply:
As an alternate idea you could delegate to the new constructor in the form of `this(env, jbytes, env->GetArrayLength(jbytes), critical)` I think.

Reply:
Done